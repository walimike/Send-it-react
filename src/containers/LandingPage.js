import React, { Component } from "react";
import NavBar from "../components/landingPage/NavBar";
import Header from "../components/landingPage/Header";
import Carousel from "../components/landingPage/Carousel";
import Footer from "../components/landingPage/Footer";
import Description from "../components/landingPage/Description";

import "../assets/styles/landing.scss";

class LandingPage extends Component {
  render() {
    return (
      <div>
        <Header/>
        <NavBar />
        <Carousel />
        <Description />
        <Footer/>
      </div>
    );
  }
}

export default LandingPage;
