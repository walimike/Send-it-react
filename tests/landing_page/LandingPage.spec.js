import React from "react";
import { shallow } from "enzyme";

import LandingPage from "../../src/containers/LandingPage";

describe("NavBar", () => {
  it("should render without crashing", () => {
    const wrapper = shallow(<LandingPage />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should open signup", () => {
    const wrapper = shallow(<LandingPage />);
    wrapper.instance().openSignUp()
    const expectedState = { openLogIn: false, openSignUp: true }
    expect(wrapper.instance().state).toEqual(expectedState);
  });

  it("should close signup", () => {
    const wrapper = shallow(<LandingPage />);
    wrapper.instance().closeSignUp()
    const expectedState = {openLogIn: false, openSignUp: false }
    expect(wrapper.instance().state).toEqual(expectedState);
  });

  it("should open login", () => {
    const wrapper = shallow(<LandingPage />);
    wrapper.instance().openLogIn()
    const expectedState = { openLogIn: true, openSignUp: false  }
    expect(wrapper.instance().state).toEqual(expectedState);
  });

  it("should close login", () => {
    const wrapper = shallow(<LandingPage />);
    wrapper.instance().closeLogIn()
    const expectedState = { openLogIn: false, openSignUp: false  }
    expect(wrapper.instance().state).toEqual(expectedState);
  });
});
