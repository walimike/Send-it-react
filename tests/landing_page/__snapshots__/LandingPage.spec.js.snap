// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render without crashing 1`] = `ShallowWrapper {}`;
